File: Java/QuickSort.java
public class QuickSort {
	private static int partitionArray(int ar[], int start, int end) {
		int pivot = ar[end];
		int pIndex = start; 
		for ( int i = start; i < end; i++ ) {
			if ( ar[i] <= pivot ) {
				int temp = ar[i]; 
				ar[i] = ar[pIndex]; 
				ar[pIndex] = temp;
			}
		}
		
		int temp = ar[end]; 
		ar[end] = ar[pIndex]; 
		ar[pIndex] = temp;
		
		return pIndex;
	}
	
	private static void quickSort(int ar[], int start, int end) {
		if ( start < end ) {
			int pIndex = partitionArray(ar, start, end);
			quickSort(ar, start, pIndex-1);
			quickSort(ar, pIndex+1, end);
		}
	}

	public static void main(String[] args) {
		int ar[] = {2, 7, 4, 1, 5, 3}; 
		int n = ar.length;
		
		System.out.print("Unsorted : ");
		for ( int i = 0; i < n; i++ ) 
			System.out.print(ar[i] + " ");
		System.out.println();
		
		quickSort(ar, 0, n-1);
		
		System.out.print("Sorted :");
		for ( int i = 0; i < n; i++ ) 
			System.out.print(ar[i] + " ");
		System.out.println();
	}

}

File: Python/Quick sort.py
def partitionArray(ar, start, end) :
	pivot = ar[end]
	pIndex = start 

	for i in range(start, end) : 
		if ar[i] <= pivot : 
			temp = ar[i]
			ar[i] = ar[pIndex]
			ar[pIndex] = temp

			pIndex += 1

	temp = ar[end]
	ar[end] = ar[pIndex]
	ar[pIndex] = temp 

	return pIndex

def quickSort(ar, start, end) : 
	if start < end : 
		pIndex = partitionArray(ar, start, end) 
		quickSort(ar, start, pIndex-1)
		quickSort(ar, pIndex+1, end)

# main 
ar = [2, 7, 4, 1, 5, 3]
n = len(ar)

print("Unsorted : ", end = " ") 
for i in range(0, n) : 
	print(ar[i], end = " ")
print()

quickSort(ar, 0, n-1)

print("Sorted : ", end = " ") 
for i in range(0, n) : 
	print(ar[i], end = " ")
print()